pkg/sessionctx/variable/sysvar.go

```diff
@@ -1915,6 +1915,10 @@ var defaultSysVars = []*SysVar{
 		s.allowTiFlashCop = TiDBOptOn(val)
 		return nil
 	}},
+	{Scope: ScopeSession, Name: TiDBEnableCommitTSOrderCheck, Value: Off, Type: TypeBool, SetSession: func(s *SessionVars, val string) error {
+		s.EnableCommitTSOrderCheck = TiDBOptOn(val)
```

Comment:
It stores the json encoded `TxnInfo`. We have to json decode to use the last commit_ts, which seems hacky.

https://github.com/pingcap/tidb/pull/57305 implements the check.

Reply:
Does this make sense?

Reply:
Since this check condition should always hold, it can always be verified without introducing a dedicated parameter. Introducing system variables or configuration items typically increases the using complexity for users.  

From an implementation perspective, introducing a separate interface to expose the `commit_ts` of the previous transaction is also costly. The `last_txn_info` cannot be directly used as it requires deserialization. 
Another possible way is to modify the `txn commit hook function` [interface](https://github.com/pingcap/tidb/blob/master/pkg/store/driver/txn/txn_driver.go#L257), allowing the KV client to set the `commit_ts` into a specific field within the session variable's statement context during the commit process like `sessVars.stmtContext.commitTS = commitTS`.

Reply:
I will try `txn commit hook function`. Thanks!

Reply:
@cfzjywxk Is this what you suggested https://github.com/pingcap/tidb/pull/57305/commits/571c73e8068dbd0344becfa618b70ade1812058c#diff-3749ef84f018df1c4d5541271a230702b5c8200c8f44268ffdff63886842f268R521?

I feel `commitTS` should be stored in the `SessionVars.LastCommitTS` instead of `StmtCtx`, because `StmtCtx` is for the current txn, while `LastCommitTS` is from the last txn.

I can refactor CommitHook to take `*TxnInfo` in a separate PR, which involves `client-go`.

Reply:
It seems that using the commit hook approach would still require modifying the `client-go` interface if we want to avoid deserializing `txn info`. 

Perhaps the original approach in [[PR #1489 of client-go](https://github.com/tikv/client-go/pull/1489)] is better, by adding a `commit_ts` interface to the transaction in `client-go` for TiDB to use. It would avoid the performance impact caused by deserializing `txn info`.

If there are future scenarios requiring access to the last transaction info, we could consider extending the newly exported `commit_ts` interface into a `last_txn_info` interface. This would allow it to return structured info instead of a string.

Reply:
sgtm

As you pointed out, this session variable is not necessary, as we should always check commit_s. I will close this PR.